### PR TITLE
Refactor setup and teardown to produce data to pass to the test suites

### PR DESCRIPTION
**Pull request by f-f**
This also:
- ports the library to the new Spago
- makes the `timeout` function safe to use given that eunit only allows for changing the timeout when directly applied to a test
- adds tests for all of this

### CODE REVIEW

File: src/Erl/Test/EUnit.purs [hunk 1]
```diff
@@ -70,17 +73,29 @@ test l t = liftF $ TestUnit l t unit
...
 
-setup :: Setup -> TestSuite -> TestSuite
-setup s su = liftF $ TestState s (pure unit) su unit
+testSuiteToForeign :: forall a. (a -> TestSuite) -> (Foreign -> TestSuite)
+testSuiteToForeign = unsafeCoerce
 
-teardown :: Teardown -> TestSuite -> TestSuite
-teardown t su = liftF $ TestState (pure unit) t su unit
+setupTeardown :: forall a. TestSetup a -> TestTeardown a -> (a -> TestSuite) -> TestSuite
+setupTeardown s t su = liftF $ TestState (map testDataToForeign s) (teardownToForeign t) (testSuiteToForeign su) unit
```
Comment (f-f) [inline src/Erl/Test/EUnit.purs:86]: Was wondering if we should call this one `setupTeardown'` so we can keep around the current implementation for when you want to ignore the runtime data

Comment (nwolverson) [reply src/Erl/Test/EUnit.purs:86]: if anything just add `setupTeardown_` for the old one to ignore

Comment (f-f) [reply src/Erl/Test/EUnit.purs:89]: Done

**Comment by f-f** [comment]
Added a more meaningful test passing the runtime data

**Comment by f-f** [comment]
Added a separate constructor for simple setup/teardown cases, see https://github.com/id3as/purescript-erl-test-eunit-discovery/pull/1

**Pull request merged**